ios/Mattermost/EventEmitterModule.m

```diff
@@ -0,0 +1,46 @@
+//
+//  EventEmitterModule.m
+//  Mattermost
+//
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+//
+
+#import "EventEmitterModule.h"
+
+@implementation EventEmitterModule
+
+RCT_EXPORT_MODULE();
+
+- (void)startObserving
+{
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(handleHardwareEnter:)
+                                               name:@"hardwareEnter"
+                                             object:nil];
+}
+
+- (void)stopObserving
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (NSArray<NSString *> *)supportedEvents
+{
+  return @[@"hardwareEnter"];
+}
+
+- (void)handleHardwareEnter:(NSNotification *)notification
+{
+  [self sendEventWithName:@"hardwareEnter"
```

Comment:
you should check if there are listeners before emitting the event

Reply:
I've added the code to check if there are listeners.